backend/src/main/java/com/example/cms/security/SecurityService.java

```diff
@@ -103,6 +103,42 @@ public boolean isForbiddenUser(User user, boolean onlyDifferentUser) {
                 .orElse(true);
     }
 
+    public boolean isForbiddenUserToDelete(User user, boolean onlyDifferentUser) {
+        return getPrincipal()
+                .map(
+                        loggedUser -> {
+                            if (onlyDifferentUser && loggedUser.getId().equals(user.getId())) {
+                                return true;
+                            } else {
+                                return isForbiddenUserToDelete(user);
+                            }
+                        })
+                .orElse(true);
+    }
+
+    public boolean isForbiddenUserToDelete(User user) {
+        return getPrincipal()
+                .map(
+                        loggedUser -> {
+                            switch (loggedUser.getAccountType()) {
+                                case ADMIN:
+                                    return false;
+                                case MODERATOR:
+                                    return !loggedUser.getId().equals(user.getId())
```

Comment:
this part is kind of hard to comprehend but that might be correlated with difficult business logic, otherwise tests look great good job

Reply:
Yeah, At the first stage of work, you need to consider what the author wanted to do, achieve and why features don't work properly xDD